sys/tools/syscalls/config.lua

```diff
@@ -0,0 +1,288 @@
+--
+-- SPDX-License-Identifier: BSD-2-Clause
+--
+-- Copyright (c) 2024 Tyler Baxter <agge@FreeBSD.org>
+-- Copyright (c) 2023 Warner Losh <imp@bsdimp.com>
+-- Copyright (c) 2019 Kyle Evans <kevans@FreeBSD.org>
+--
+
+-- Code to read in the config file that drives this. Since we inherit from the
+-- FreeBSD makesyscall.sh legacy, all config is done through a config file that
+-- sets a number of varibale (as noted below, it used to be a .sh file that was
+-- sourced in. This dodges the need to write a command line parser.
+
+local util = require("tools/util")
+
+--
+-- Global config map.
+-- Default configuration is native (amd64). Any of these may get replaced by an
+-- optionally specified configuration file.
+--
+local config = {
+    sysnames = "syscalls.c",
+    syshdr = "../sys/syscall.h",
+    sysmk = "/dev/null",
+    syssw = "init_sysent.c",
+    systrace = "systrace_args.c",
+    sysproto = "../sys/sysproto.h",
+    libsysmap = "/dev/null",
+    libsys_h = "/dev/null",
+    sysproto_h = "_SYS_SYSPROTO_H_",
+    syscallprefix = "SYS_",
+    switchname = "sysent",
+    namesname = "syscallnames",
+    abi_flags = {},
+    abi_func_prefix = "",
+    abi_type_suffix = "",
+    abi_long = "long",
+    abi_u_long = "u_long",
+    abi_semid_t = "semid_t",
+    abi_size_t = "size_t",
+    abi_ptr_array_t = "",
+    abi_headers = "",
+    abi_intptr_t = "intptr_t",
+    ptr_intptr_t_cast = "intptr_t",
+    obsol = {},
+    unimpl = {},
+    capabilities_conf = "capabilities.conf",
```

Comment:
Support for capabilities.conf should be dropped as the last consumer went away with be67ea40c5a0fd375d754a92e6cd8a3bfc508ba3. It should provably not be removed as part of the refactor, but done on one side or the other.